### ISSUE TITLE
[Windows] Cannot build stable releaes e.g. 5.2.1 - documentation should be updated

### ISSUE THREAD

**Issue by gitrpbrme** [owncloud/client#11616]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the bug

Hi Team,
While following [official documentation](https://doc.owncloud.com/desktop/next/appendices/building.html), at the **Windows Build The Desktop App** Step, I have the option to select a branch to be built.

The example on the documentation does not work:

`You can find the git checkout in C:\CraftRoot\downloads\git\owncloud\owncloud-client. There you can use the usual git commands to switch branches and remotes, e.g., to build the 5.2 stable branch you can use craft with --set version parameter:

git checkout 5.2
craft --set version=5.2 owncloud-client`

Can you please advise on how should I do to build a stable release instead of the master release?

### Expected behavior

I expected to be able to change to stable release to build it.
The master release when I built, the .exe gets generated, but when I run it on the logs show the below, so I wanted to build a stable release:

ownCloud 6.0.0-git d872c3 Bibliotecas Qt 6.7.0, OpenSSL 3.2.1 30 Jan 2024 Usando o plug-in de arquivos virtuais: suffix OS: windows-10.0.19045 (build arch: x86_64, CPU arch: x86_64) QPA: windows ownCloud
24-04-18 11:47:32:390 [ debug default ]	[ OCC::Logger::setLogRules ]:	""
24-04-18 11:47:32:390 [ info gui.main ]:	################## "ownCloud" locale: "pt_BR" version: "ownCloud 6.0.0-git d872c3 Bibliotecas Qt 6.7.0, OpenSSL 3.2.1 30 Jan 2024 Usando o plug-in de arquivos virtuais: suffix OS: windows-10.0.19045 (build arch: x86_64, CPU arch: x86_64) QPA: windows"
24-04-18 11:47:32:390 [ info gui.main ]:	Arguments: QList("C:\\CraftRoot\\bin\\owncloud.exe", "--logfile", "C:\\temp\\owncloud.log")
24-04-18 11:47:32:468 [ info gui.account.manager ]:	Restored:  0  unknown certs.
24-04-18 11:47:32:468 [ info gui.folder.manager ]:	Setup folders from settings file
24-04-18 11:47:32:484 [ critical sync.plugins ]:	**Could not load plugin: not existant or bad metadata "ownCloud_vfs_wincfapi"**
24-04-18 11:47:32:484 [ fatal default ]:	Could not load plugin

### Steps to reproduce the issue

1. Follow the build documentation from Windows 
2. On the "Build The Desktop App", sub step:
You can find the git checkout in C:\CraftRoot\downloads\git\owncloud\owncloud-client. There you can use the usual git commands to switch branches and remotes, e.g., to build the 5.2 stable branch you can use craft with --set version parameter:

git checkout 5.2
craft --set version=5.2 owncloud-client

3. When I run the below, I get the error:
craft --configure --make --install
craft owncloud-client

PS C:\CraftRoot> craft owncloud-client
Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
C:\CraftRoot\etc\blueprints\locations\craft-blueprints-owncloud\owncloud\owncloud-client\owncloud-client.py failed:
You defined an invalid target 5 for owncloud/owncloud-client, avaialble versions are **['2.5.4.11654', 'master', '2.6', '2.7', '2.8', '2.9', '2.10']**

### Client version number

Latest build

**Comment by TheOneRing** [comment]
Please follow the steps described in https://github.com/owncloud/ownbuild/blob/master/README.md
We have to decide what will happen to the other doc

**Comment by gitrpbrme** [comment]
Thanks for the help Hanna. I was able to use the ownbuild.py.
I was able to generate the same build with the same problem, although...

**Now, I was able to change and start the build process to the latest stable branch, by running the following command:**

`py.exe  ownbuild.py --branch 5  --target windows-cl-msvc2022-x86_64 -- --set revision=v5.2.1 owncloud-client`

But after a while compiling, I got the following:

```
PS C:\CraftRoot> py.exe  ownbuild.py --branch 5  --target windows-cl-msvc2022-x86_64 -- --set revision=v5.2.1 owncloud-client
C:\python36\python.exe C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\5\.craft.ini --variables Root=C:\CraftRoot\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c --unshelve C:\CraftRoot\5\.craft.shelf
CraftMaster Arguments: C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\5\.craft.ini --variables Root=C:\CraftRoot\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c --unshelve C:\CraftRoot\5\.craft.shelf
Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
===============================================================================================================================================================================================================================================================================
*** Handling package: add-bluprints-template, action: fetch ***
*** Action: fetch for add-bluprints-template ***
*** Craft fetch succeeded: add-bluprints-template after 0s ***
C:\CraftRoot\craft\internal_blueprints\add-bluprints-template\add-bluprints-template.py failed:
Failed to load package C:\CraftRoot\craft\internal_blueprints\add-bluprints-template\add-bluprints-template.py
Command C:\python36\python.exe -X utf8 -u C:\CraftRoot\5\windows-cl-msvc2022-x86_64\craft\bin\craft.py --unshelve C:\CraftRoot\5\.craft.shelf failed with exit code: 1
Failed to setup craft

```

**Those files exists and look good. Am I missing something here? I just wanted to build a clean latest release from scratch**

Thanks for all your help.

Hopefully you can help me here.

Best regards,

**Comment by TheOneRing** [comment]
Can you try to perform the first call without any args and once it succeeded try the --set call?

**Comment by gitrpbrme** [comment]
> Can you try to perform the first call without any args and once it succeeded try the --set call?

With no args, it doesn't run:
```
PS C:\CraftRoot> py.exe  ownbuild.py  owncloud-client
C:\python36\python.exe C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\master\.craft.ini --variables Root=C:\CraftRoot\master CiBuild=False CreateCache=False --target windows-msvc2019_64-cl -c --unshelve C:\CraftRoot\master\.craft.shelf
CraftMaster Arguments: C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\master\.craft.ini --variables Root=C:\CraftRoot\master CiBuild=False CreateCache=False --target windows-msvc2019_64-cl -c --unshelve C:\CraftRoot\master\.craft.shelf
Target windows-msvc2019_64-cl is not a valid target. Valid targets are ['windows-cl-msvc2022-x86_64', 'windows-cl-msvc2022-x86_64-debug']
Failed to setup craft
```

So I have to at least set the target (with the master branch it builds successfully, but I have the same error at runtime - trayicon doesn't show and show the error on the logs):

```
PS C:\CraftRoot> py.exe  ownbuild.py --target windows-cl-msvc2022-x86_64 owncloud-client
C:\python36\python.exe C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\master\.craft.ini --variables Root=C:\CraftRoot\master CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 --setup -c --unshelve C:\CraftRoot\master\.craft.shelf
CraftMaster Arguments: C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\master\.craft.ini --variables Root=C:\CraftRoot\master CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 --setup -c --unshelve C:\CraftRoot\master\.craft.shelf
Cloning into 'C:\CraftRoot\master\craft-clone'...
remote: Enumerating objects: 117846, done.
remote: Counting objects: 100% (4060/4060), done.
remote: Compressing objects: 100% (745/745), done.
remote: Total 117846 (delta 3975), reused 3315 (delta 3315), pack-reused 113786
Receiving objects: 100% (117846/117846), 33.69 MiB | 7.96 MiB/s, done.
Resolving deltas: 100% (79286/79286), done.
Junção criada para C:\CraftRoot\master\windows-cl-msvc2022-x86_64\craft <<===>> C:\CraftRoot\master\craft-clone
Generate Settings
Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
========================================================================================================================
*** Handling package: add-bluprints-template, action: fetch ***
*** Action: fetch for add-bluprints-template ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft fetch succeeded: add-bluprints-template after 1s ***
========================================================================================================================
*** Handling package: add-bluprints-template, action: fetch ***
*** Action: fetch for add-bluprints-template ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft fetch succeeded: add-bluprints-template after 1s ***
Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
setOptions: BlueprintSettings.ini [virtual]ignored = True
setOptions: BlueprintSettings.ini [libs]ignored = True
setOptions: BlueprintSettings.ini [dev-utils]ignored = True
========================================================================================================================
*** Handling package: craft/craft-core, action: update ***
*** Action: fetch-binary for craft/craft-core ***
*** craft/craft-core not found in cache ***
*** Action: update for craft/craft-core ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft update succeeded: craft/craft-core after 1s ***
========================================================================================================================
*** Handling package: craft/craft-blueprints-kde, action: update ***
*** Action: fetch-binary for craft/craft-blueprints-kde ***
*** craft/craft-blueprints-kde not found in cache ***
*** Action: update for craft/craft-blueprints-kde ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft update succeeded: craft/craft-blueprints-kde after 1s ***
========================================================================================================================
*** Handling package: craft/craft-blueprints-owncloud, action: update ***
*** Action: fetch-binary for craft/craft-blueprints-owncloud ***
*** craft/craft-blueprints-owncloud not found in cache ***
*** Action: update for craft/craft-blueprints-owncloud ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft update succeeded: craft/craft-blueprints-owncloud after 1s ***



C:\python36\python.exe C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\master\.craft.ini --variables Root=C:\CraftRoot\master CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c owncloud-client
CraftMaster Arguments: C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\master\.craft.ini --variables Root=C:\CraftRoot\master CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c owncloud-client
Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
*** owncloud/owncloud-client is up to date, nothing to do ***
```

Now, if I run the command with the proper branch and revision (latest stable), here is what I get (error in cacert.py):

```
PS C:\CraftRoot> py ownbuild.py --branch 5  --target windows-cl-msvc2022-x86_64 -- --set revision=v5.2.1 owncloud-client
C:\python36\python.exe C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\5\.craft.ini --variables Root=C:\CraftRoot\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 --setup -c --unshelve C:\CraftRoot\5\.craft.shelf
CraftMaster Arguments: C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\5\.craft.ini --variables Root=C:\CraftRoot\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 --setup -c --unshelve C:\CraftRoot\5\.craft.shelf
Cloning into 'C:\CraftRoot\5\craft-clone'...
remote: Enumerating objects: 117846, done.
remote: Counting objects: 100% (4060/4060), done.
remote: Compressing objects: 100% (745/745), done.
remote: Total 117846 (delta 3973), reused 3315 (delta 3315), pack-reused 113786
Receiving objects: 100% (117846/117846), 33.69 MiB | 7.90 MiB/s, done.
Resolving deltas: 100% (79284/79284), done.
Junção criada para C:\CraftRoot\5\windows-cl-msvc2022-x86_64\craft <<===>> C:\CraftRoot\5\craft-clone
Generate Settings
Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
========================================================================================================================
*** Handling package: add-bluprints-template, action: fetch ***
*** Action: fetch for add-bluprints-template ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft fetch succeeded: add-bluprints-template after 1s ***
========================================================================================================================
*** Handling package: add-bluprints-template, action: fetch ***
*** Action: fetch for add-bluprints-template ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft fetch succeeded: add-bluprints-template after 0s ***
Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
setOptions: BlueprintSettings.ini [virtual]ignored = True
setOptions: BlueprintSettings.ini [libs]ignored = True
setOptions: BlueprintSettings.ini [dev-utils]ignored = True
========================================================================================================================
*** Handling package: craft/craft-core, action: update ***
*** Action: fetch-binary for craft/craft-core ***
*** craft/craft-core not found in cache ***
*** Action: update for craft/craft-core ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch --tags
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout ab3623cbeeabe3444742a3325a201d73062e301c
Note: switching to 'ab3623cbeeabe3444742a3325a201d73062e301c'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at ab3623cbe Fix boost build system for ARM
*** Craft update succeeded: craft/craft-core after 1s ***
========================================================================================================================
*** Handling package: craft/craft-blueprints-kde, action: update ***
*** Action: fetch-binary for craft/craft-blueprints-kde ***
*** craft/craft-blueprints-kde not found in cache ***
*** Action: update for craft/craft-blueprints-kde ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch --tags
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout d2f855b1fa83f05025cc342c931d0805a5025551
Note: switching to 'd2f855b1fa83f05025cc342c931d0805a5025551'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at d2f855b1 [KMyMoney] try to fix Appimage build
*** Craft update succeeded: craft/craft-blueprints-kde after 2s ***
========================================================================================================================
*** Handling package: craft/craft-blueprints-owncloud, action: all ***
*** Action: fetch-binary for craft/craft-blueprints-owncloud ***
*** craft/craft-blueprints-owncloud not found in cache ***
*** Action: fetch for craft/craft-blueprints-owncloud ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout 5
Switched to a new branch '5'
branch '5' set up to track 'origin/5'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout 5
Already on '5'
Your branch is up to date with 'origin/5'.
*** Action: unpack for craft/craft-blueprints-owncloud ***
*** Action: configure for craft/craft-blueprints-owncloud ***
*** Action: make for craft/craft-blueprints-owncloud ***
*** Action: cleanimage for craft/craft-blueprints-owncloud ***
*** Action: install for craft/craft-blueprints-owncloud ***
*** Action: post-install for craft/craft-blueprints-owncloud ***
*** Action: qmerge for craft/craft-blueprints-owncloud ***
*** Action: post-qmerge for craft/craft-blueprints-owncloud ***
*** Craft all succeeded: craft/craft-blueprints-owncloud after 1s ***



Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
C:\CraftRoot\craft\blueprints\core\cacert\cacert.py failed:
Failed to load package C:\CraftRoot\craft\blueprints\core\cacert\cacert.py
Command C:\python36\python.exe -X utf8 -u C:\CraftRoot\5\windows-cl-msvc2022-x86_64\craft\bin\craft.py --unshelve C:\CraftRoot\5\.craft.shelf failed with exit code: 1
Failed to setup craft
```

If I just rerun this command, now I have the error I mentioned earlier:

```
Command C:\python36\python.exe -X utf8 -u C:\CraftRoot\5\windows-cl-msvc2022-x86_64\craft\bin\craft.py --unshelve C:\CraftRoot\5\.craft.shelf failed with exit code: 1
Failed to setup craft
PS C:\CraftRoot> py ownbuild.py --branch 5  --target windows-cl-msvc2022-x86_64 -- --set revision=v5.2.1 owncloud-client
C:\python36\python.exe C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\5\.craft.ini --variables Root=C:\CraftRoot\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c --unshelve C:\CraftRoot\5\.craft.shelf
CraftMaster Arguments: C:\CraftRoot\craftmast\CraftMaster.py --config C:\CraftRoot\5\.craft.ini --variables Root=C:\CraftRoot\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c --unshelve C:\CraftRoot\5\.craft.shelf
Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
========================================================================================================================
*** Handling package: add-bluprints-template, action: fetch ***
*** Action: fetch for add-bluprints-template ***
*** Craft fetch succeeded: add-bluprints-template after 0s ***
C:\CraftRoot\craft\internal_blueprints\add-bluprints-template\add-bluprints-template.py failed:
Failed to load package C:\CraftRoot\craft\internal_blueprints\add-bluprints-template\add-bluprints-template.py
Command C:\python36\python.exe -X utf8 -u C:\CraftRoot\5\windows-cl-msvc2022-x86_64\craft\bin\craft.py --unshelve C:\CraftRoot\5\.craft.shelf failed with exit code: 1
Failed to setup craft
```

**Comment by gitrpbrme** [comment]
Hi,
In summary, I was able to create a build without --set call.

Afterwards, I tried with the --set arguments, it failed at 
```

Craft               : C:\CraftRoot
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\CraftRoot\download
Cache repository    : https://files.kde.org/craft/Qt6/24.04/windows/cl/msvc2022/x86_64
C:\CraftRoot\craft\blueprints\core\cacert\cacert.py failed:
Failed to load package C:\CraftRoot\craft\blueprints\core\cacert\cacert.py
```

I ran the command once again and now the error change to:
```*** Craft fetch succeeded: add-bluprints-template after 0s ***
C:\CraftRoot\craft\internal_blueprints\add-bluprints-template\add-bluprints-template.py failed:
Failed to load package C:\CraftRoot\craft\internal_blueprints\add-bluprints-template\add-bluprints-template.py
Command C:\python36\python.exe -X utf8 -u C:\CraftRoot\master\windows-cl-msvc2022-x86_64\craft\bin\craft.py --unshelve C:\CraftRoot\master\.craft.shelf failed with exit code: 1
```

**Last but not least, I now tried to run the build again without --set call, now it gives me the last error all the time.
The only way is to remove craftroot folder and reinstall.

Have you ever faced this while setting up the Craft package and setting up a build?**

**Comment by TheOneRing** [comment]
It appears you are mixing a Craft setup with a ownBuild setup.
While Craft is used in any case its not supported to run ownBuild from inside a craft environment.
Please use a fresh terminal window with a clean environment.

**Comment by gitrpbrme** [comment]
> It appears you are mixing a Craft setup with a ownBuild setup. While Craft is used in any case its not supported to run ownBuild from inside a craft environment. Please use a fresh terminal window with a clean environment.

Indeed I was doing wrong with ownbuild.py, so I've setup a new folder to run it.

The latest release still doesn't run the .exe, same error as in the beginning so I guess it's something to do with my build environment (Win10 X64).

By running the command below, the only thing that doesn't get generated is the owncloud.exe file 😔 

Not sure if I'm missing to run anything afterwards.
```
C:\ownbuild>py ownbuild.py --branch 5 --target  windows-cl-msvc2022-x86_64  -- --set revision=v5.2.1 owncloud-client
C:\python36\python.exe C:\ownbuild\craftmast\CraftMaster.py --config C:\ownbuild\5\.craft.ini --variables Root=C:\ownbuild\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c --unshelve C:\ownbuild\5\.craft.shelf
CraftMaster Arguments: C:\ownbuild\craftmast\CraftMaster.py --config C:\ownbuild\5\.craft.ini --variables Root=C:\ownbuild\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c --unshelve C:\ownbuild\5\.craft.shelf
Craft               : C:\ownbuild\5\windows-cl-msvc2022-x86_64
Version             : master
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\ownbuild\5\downloads
========================================================================================================================*** Handling package: add-bluprints-template, action: fetch ***
*** Action: fetch for add-bluprints-template ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Previous HEAD position was d2f855b1 [KMyMoney] try to fix Appimage build
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft fetch succeeded: add-bluprints-template after 12 seconds ***
========================================================================================================================*** Handling package: add-bluprints-template, action: fetch ***
*** Action: fetch for add-bluprints-template ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft fetch succeeded: add-bluprints-template after 1 second ***
Craft               : C:\ownbuild\5\windows-cl-msvc2022-x86_64
Version             : master
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\ownbuild\5\downloads
setOptions: BlueprintSettings.ini [virtual]ignored = True
========================================================================================================================*** Handling package: craft/craft-core, action: update ***
*** Action: fetch-binary for craft/craft-core ***
*** craft/craft-core not found in cache ***
*** Action: update for craft/craft-core ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch --tags
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout ab3623cbeeabe3444742a3325a201d73062e301cHEAD is now at ab3623cbe Fix boost build system for ARM
*** Craft update succeeded: craft/craft-core after 11 seconds ***
========================================================================================================================*** Handling package: craft/craft-blueprints-kde, action: update ***
*** Action: fetch-binary for craft/craft-blueprints-kde ***
*** craft/craft-blueprints-kde not found in cache ***
*** Action: update for craft/craft-blueprints-kde ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch --tags
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout d2f855b1fa83f05025cc342c931d0805a5025551Note: switching to 'd2f855b1fa83f05025cc342c931d0805a5025551'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at d2f855b1 [KMyMoney] try to fix Appimage build
*** Craft update succeeded: craft/craft-blueprints-kde after 12 seconds ***
========================================================================================================================*** Handling package: craft/craft-blueprints-owncloud, action: update ***
*** Action: fetch-binary for craft/craft-blueprints-owncloud ***
*** craft/craft-blueprints-owncloud not found in cache ***
*** Action: update for craft/craft-blueprints-owncloud ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout 5
Switched to branch '5'
Your branch is up to date with 'origin/5'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout 5
Already on '5'
Your branch is up to date with 'origin/5'.
*** Craft update succeeded: craft/craft-blueprints-owncloud after 1 second ***
========================================================================================================================*** Handling package: craft/craftmaster, action: update ***
*** Action: fetch-binary for craft/craftmaster ***
*** craft/craftmaster not found in cache ***
*** Action: update for craft/craftmaster ***
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always fetch
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
executing command: C:\Program Files\Git\cmd\git.exe -c color.ui=always checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
*** Craft update succeeded: craft/craftmaster after 11 seconds ***



Craft               : C:\ownbuild\5\windows-cl-msvc2022-x86_64
Version             : master
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\ownbuild\5\downloads
*** core/cacert is up to date, nothing to do ***
*** libs/runtime is up to date, nothing to do ***
*** dev-utils/cmake-base is up to date, nothing to do ***
*** dev-utils/kshimgen is up to date, nothing to do ***
*** dev-utils/cmake is up to date, nothing to do ***
*** python-modules/pip-system is up to date, nothing to do ***
*** python-modules/virtualenv is up to date, nothing to do ***
*** dev-utils/python3 is up to date, nothing to do ***
*** python-modules/pip is up to date, nothing to do ***
*** dev-utils/patch is up to date, nothing to do ***
*** dev-utils/jom is up to date, nothing to do ***
*** dev-utils/ninja is up to date, nothing to do ***
*** craft/craft-core is up to date, nothing to do ***
*** craft/craft-blueprints-kde is up to date, nothing to do ***
*** craft/craft-blueprints-owncloud is up to date, nothing to do ***
*** dev-utils/icoutils is up to date, nothing to do ***
*** dev-utils/nasm is up to date, nothing to do ***
*** dev-utils/perl is up to date, nothing to do ***
*** dev-utils/pkg-config is up to date, nothing to do ***
*** dev-utils/symsorter is up to date, nothing to do ***
*** libs/zlib is up to date, nothing to do ***
*** libs/openssl is up to date, nothing to do ***
*** libs/brotli is up to date, nothing to do ***
*** libs/libzstd is up to date, nothing to do ***
*** libs/libpng is up to date, nothing to do ***
*** libs/libjpeg-turbo is up to date, nothing to do ***
*** libs/sqlite is up to date, nothing to do ***
*** libs/libbzip2 is up to date, nothing to do ***
*** libs/pcre2 is up to date, nothing to do ***
*** libs/freetype is up to date, nothing to do ***
*** libs/harfbuzz is up to date, nothing to do ***
*** libs/qt6/qtbase is up to date, nothing to do ***
*** libs/qt6/qtshadertools is up to date, nothing to do ***
*** libs/qt6/qtdeclarative is up to date, nothing to do ***
*** libs/qt6/qttools is up to date, nothing to do ***
*** libs/qt6/qttranslations is up to date, nothing to do ***
*** kde/frameworks/extra-cmake-modules is up to date, nothing to do ***
*** libs/giflib is up to date, nothing to do ***
*** libs/kdsingleapplication is up to date, nothing to do ***
*** libs/liblzma is up to date, nothing to do ***
*** libs/libre-graph-api-cpp-qt-client is up to date, nothing to do ***
*** libs/libxml2 is up to date, nothing to do ***
*** libs/nlohmann-json is up to date, nothing to do ***
*** libs/tiff is up to date, nothing to do ***
*** libs/webp is up to date, nothing to do ***
*** libs/qt6/qtimageformats is up to date, nothing to do ***
*** libs/qt6/qtscxml is up to date, nothing to do ***
*** libs/qt6/qtsvg is up to date, nothing to do ***
*** owncloud/libcrashreporter-qt is up to date, nothing to do ***
*** qt-libs/qtkeychain is up to date, nothing to do ***



C:\python36\python.exe C:\ownbuild\craftmast\CraftMaster.py --config C:\ownbuild\5\.craft.ini --variables Root=C:\ownbuild\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c --set revision=v5.2.1 owncloud-client
CraftMaster Arguments: C:\ownbuild\craftmast\CraftMaster.py --config C:\ownbuild\5\.craft.ini --variables Root=C:\ownbuild\5 CiBuild=False CreateCache=False --target windows-cl-msvc2022-x86_64 -c --set revision=v5.2.1 owncloud-client
Craft               : C:\ownbuild\5\windows-cl-msvc2022-x86_64
Version             : master
ABI                 : windows-cl-msvc2022-x86_64
Download directory  : C:\ownbuild\5\downloads
[owncloud/owncloud-client]
revision=v5.2.1
```

If I run without the --set revision, it builds properly.
However, when I run the owncloud.exe, it gives me the error popup:

**Some settings were configured in newer versions of this client and use features that are not available in this version**

So, bottom line, still not possible to generate a latest stable release from my side.

- If I use the Linux, can I generate the .exe files?
- Should I use a Windows 11 installation as maybe craft and etc it's not compatible with Win10?

**Comment by gitrpbrme** [comment]
I recreated the environment on another server, it worked. Thanks.
